syfertext/encoders/bert_encoder.py

```diff
@@ -0,0 +1,16 @@
+from typing import Dict, List
+from transformers import BertTokenizer
+
+class BERTEncoder:
+    
+    def __init__(self):
+        self.tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')
+        
+    def __call__(self, text:List) -> Dict:
+        inputs = self.tokenizer(text)
+        return {"token_ids": inputs["input_ids"]}
+
+    @property
+    def tokenizer_ref(self):
```

Comment:
Can you please explain this to me?

Reply:
The `TextReader` class gets the `token_ids` key from the encoding output, which is the list of integers representing the tokens in the vocabulary. However, the HuggingFace tokenizer calls this key of the encoding output `input_ids` (which will return the list of integers). In order for this encoder to work with the `Reader`, I needed to map the `input_ids` key to the `token_ids` key, so I returned a dictionary as the encoder output.

Reply:
@AlanAboudib this is just so we can get a reference to the tokenizer when collating the batches, which requires this tokenizer to use. Rather than getting it as an attribute from the encoder, where it could still be modified (ex. `encoder.tokenizer = __` ), I created a property so that the tokenizer attribute of the class cannot be changed in a program, it can only be accessed. I can always remove this and just use the way of accessing as an attribute from the class if you think this is unnecessary.